nessie/src/test/java/org/apache/iceberg/nessie/TestNessieIcebergClient.java

```diff
@@ -91,6 +109,361 @@ public void testWithReferenceAfterRecreatingBranch()
     Assertions.assertThat(client.withReference(branch, null)).isNotEqualTo(client);
   }
 
+  @Test
```

Comment:
this is a fairly long test method. I would suggest to split it up into multiple smaller test methods